packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -95,13 +95,15 @@ export default declare((api, opts) => {
     const props = path.node.properties;
     const keys = [];
     let allLiteral = true;
+    let hasTemplateLiteral = false;
 
     for (const prop of props) {
       if (t.isIdentifier(prop.key) && !prop.computed) {
         // since a key {a: 3} is equivalent to {"a": 3}, use the latter
         keys.push(t.stringLiteral(prop.key.name));
       } else if (t.isTemplateLiteral(prop.key)) {
         keys.push(t.cloneNode(prop.key));
+        hasTemplateLiteral = true;
```

Comment:
This flag is kinda redundant, you could just set `allLiteral = false` (becase `TemplateLiteral` is most often not literal). Or remove this branch, replace the following condition's `isLiteral` with `isImmutable`, and then `TemplateLiteral` would fall through to the `else`.

Reply:
No, because when `allLiteral` is `false` we need to inject the `toPropertyKey` helper. We don't need it if we have template literals.